web/src/ProductSelectionPage.jsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useInstallerClient } from "./context/installer";
+import { useSoftware } from "./context/software";
+
+import {
+  Button,
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  Radio
+} from "@patternfly/react-core";
+
+import {
+  EOS_PRODUCT_SUBSCRIPTIONS as SectionIcon,
+} from "eos-icons-react";
+
+import Layout from "./Layout";
+import Center from "./Center";
+
+function ProductSelectionPage() {
+  const client = useInstallerClient();
+  const navigate = useNavigate();
+  const { products, selectedProduct } = useSoftware();
+  const previous = selectedProduct?.id;
+  const [selected, setSelected] = useState(selectedProduct?.id);
+
+  useEffect(() => {
+    // TODO: display a notification in the UI to emphasizes that
+    // selected product has changed
+    return client.software.onProductChange(setSelected);
+  }, [client.software]);
+
+  const isSelected = p => p.id === selected;
+
+  const accept = () => {
+    if (selected === previous) {
+      navigate("/");
+      return;
+    }
+
+    // TODO: handle errors
+    client.software.selectProduct(selected)
+      .then(() => navigate("/"));
+  };
+
+  const SelectButton = () => {
+    return (
+      <Button isLarge variant="primary" onClick={accept}>
+        Select
+      </Button>
+    );
+  };
+
+  // FIXME: improve this
+  if (!products) {
+    return (
+      <Layout
+        sectionTitle="D-Installer"
+        SectionIcon={SectionIcon}
+        FooterActions={SelectButton}
+      >
+        <Center>Loading available products...</Center>
+      </Layout>
+    );
+  }
```

Comment:
I guess this can be changed. What about

```suggestion
  if (!products) return <LoadingEnvironment text="Loading available products, please wait..." />
```

Of course, it's needed to import the LoadingEnvironment component (or simply Loading if we endup renaming it)

Reply:
Note that makes no sense to show the "SelectButton" action while "Loading available products"